### PR TITLE
Deprecate the backslash operator

### PR DESCRIPTION
**Pull request by jhpalmieri**
Deprecate the backslash operator.

Deprecate the backslash operator: it is underused and its current implementation conflicts with standard Python usage of `\`. A [poll on sage-devel ](https://groups.google.com/g/sage-devel/c/8G-tPMdtNX4/m/tRCfUVHiCAAJ) supports this decision.

The backslash operator is only used in a few places, mainly matrices and matroids; those uses are deprecated.

### :memo: Checklist

- [X] The title is concise, informative, and self-explanatory.
- [X] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [X] I have created tests covering the changes.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: src/doc/ru/tutorial/tour_linalg.rst [hunk 1]
```diff
@@ -37,15 +37,6 @@ Sage поддерживает стандартные конструкции из
...
-``A \ Y`` вместо ``A.solve_right(Y)``.
-
-.. link
-
-::
-
-    sage: A \ Y
-    (-2, 1, 0)
 
 Если решения не существует, то Sage вернет ошибку:
```
Comment (kwankyu) [inline src/doc/ru/tutorial/tour_linalg.rst:40]: You may delete empty line 40?

**Comment by kwankyu** [comment]
Otherwise, lgtm.

File: src/doc/ru/tutorial/tour_linalg.rst (hunk 1 above)
Comment (jhpalmieri) [reply src/doc/ru/tutorial/tour_linalg.rst:40]: Done.

**Review by kwankyu — APPROVED** [review]
Thanks.

**Comment by jhpalmieri** [comment]
Thank you!

**Pull request merged**